src/main/org/apache/tools/bzip2/BlockSort.java

```diff
@@ -894,8 +893,8 @@ private void mainQSort3(final CBZip2OutputStream.Data dataShadow,
                     stack_ll[sp] = m;
                     stack_hh[sp] = hi;
                     stack_dd[sp] = d;
-                    sp++;
                 }
+                sp++;
```

Comment:
technically this is redundant code but as `sp` gets incremented several time inside of the else block I'd prefer to leave the code as it. It makes shape of the code blocks that assign to the stack arrays the same for all four occasions (the if and three times inside of the else block).